concierge_scheduler/concierge_zabbix.py

```diff
@@ -307,32 +380,96 @@ def import_actions(self):
             # "pyzabbix.ZabbixAPIException: ('Error -32500: Application error.,
             # No permissions to referred object or it does not exist!', -32500)"
             _warn(str(err))
+        _info('Importing trigger actions')
         self.import_trigger_actions()
+        _info('Importing registration actions')
         self.import_registration_actions()
 
-    def import_media_types(self, component):
+    def compare_ids(self, component):
+        """
+        Compare the ids and hashes of components for imported configuration.
+        Remove any components that are in the destination server but not in the imported configuration
+        :param component: Which component to compare the data of
         """
-        import backed up media
 
-        :param component: what media to import
-        :return:
+        component_id = self.id_mapping[component]['id']
+        ids_to_delete = []
+        for item in self.original_ids[component]:
+            import_id = item[component_id]
+            if import_id in self.dest_ids[component]:
+                if item["hash"] != self.dest_ids[component][import_id][1]:
+                    ids_to_delete.append(import_id)
+
+        for del_id in list(ids_to_delete):
+            print("Deleting '{}' ids: {}".format(component, del_id))
+            getattr(self.zbx_client, self.id_mapping[component]['api_name']).delete(del_id)
+
+    def import_components(self, component):
+        """
+        Method for importing components which cannot be configured using a configuration import.
+        If the component exists in destination server it is updated
+        :param component: Component to import
         """
-        # ToDo: done because default DB install creates some mediatypes.
-        # We could run self.zbx_client.mediatype.update() instead, then handle
-        # handle the exception and only run create() if it fails
-        # 
-        self.zbx_client.mediatype.delete("1", "2", "3")
         import_file = '{}/{}.json'.format(self.data_dir, component)
-        with open(import_file, 'r') as f:
-            component_data = f.read()
-            mediatypes = json.loads(component_data)
-            for mediatype in mediatypes:
+        component_method = self.id_mapping[component]['api_name']
+        file = open(import_file).read()
+        if json.loads(file):
```

Comment:
what are you trying to check here because you've got json.loads twice? If it is the existent of the file there is a path.exists method for that.

Reply:
changed to `if os.path.isfile(file):`. I had originally put that if statement because it errored, but changed it to checking if the file exists